src/scripts/stores/user-store.jsx

```diff
@@ -16,16 +16,22 @@ const UserStore = assign({}, EventEmitter.prototype, {
     this.removeListener("change", cb);
   },
 
-  handleIdToken(accessToken, idToken) {
-    resoundAPI.users.create(accessToken, idToken);
+  handleIdToken(accessToken, idToken, tenant) {
+    return resoundAPI.users.create(accessToken, idToken, tenant);
   }
 });
 
 AppDispatcher.register(action => {
   switch (action.actionType) {
     case "USER_LOGGED_IN":
-      UserStore.emitChange();
-      UserStore.handleIdToken(action.accessToken, action.idToken);
+      UserStore.handleIdToken(
+        action.accessToken,
+        action.idToken,
+        action.tenant
+      ).then(() => {
+        UserStore.emitChange();
+        window.location.reload();
+      });
```

Comment:
You're right! This bug must not have come up b/c our apps have low latency to each other.  One thing though, you probably don't need line 33: `window.location.reload();` since `Header.jsx` is listening UserStore: `UserStore.addChangeListener(this._redirect);`, which will redirect to `/` if logged in.  Is that what you're trying to do with `window.location.reload();`?

Reply:
Well, I'm not that familiar with react, then you will answer me :grimacing: 

If I remove the `window.location.reload()` code.
All the requests to `audio` endpoint, started to get `unauthenticated`
For example the ones dispatched by this component https://github.com/ProjectResound/store-upload/blob/master/src/scripts/app.jsx#L41

Digging out, I found that this `const` https://github.com/ProjectResound/store-upload/blob/master/src/scripts/services/resound-api.js#L5 was empty, so re reload, kind of force that to have a value, I think only the redirect doesn't have the same result.

Also that reload, existed before https://github.com/ProjectResound/store-upload/pull/28/files#diff-d775e44f53bf4832b5a0815dd1fe2adeL24. I think that problem was happening before, tha's why we were using it.

Reply:
I see what you mean now. It happens during the first loading of resound right after authenticating. Yes, it makes sense. Let's put this change into `jonatasrancan:multi-tenancy` branch too and plan to merge that into master.